### PR TITLE
GH-300: Configurable cmsComponentProviders

### PR DESCRIPTION
**Pull request by dunqan**

### CODE REVIEW

File: projects/storefrontlib/src/lib/cms/cms-module-config.ts [hunk 1]
```diff
@@ -34,6 +41,7 @@ export abstract class CmsModuleConfig extends OccConfig
   };
 
   cmsComponentMapping?: CMSComponentMappingConfig;
+  cmsComponentProviders?: CMSComponentProviders;
```
Comment (tobi-or-not-tobi) [inline projects/storefrontlib/src/lib/cms/cms-module-config.ts]: One of the ideas that I had earlier this week (forgot to tell), is to combine the configuration of the component mapping and component providers in one object. The config would have a typing that allows for both as an optional property. I believe it simplifies configuration as well as its' all more connected when you customise.

File: projects/core/src/site-context/facade/language.service.ts [hunk 2]
```diff
@@ -34,6 +34,11 @@ export class LanguageService {
     this.store.dispatch(new LoadLanguages());
   }
 
+  /**
```
Comment (tobi-or-not-tobi) [inline projects/core/src/site-context/facade/language.service.ts]: nice :)

File: projects/storefrontlib/src/lib/cms-lib/search-box/search-box.component.spec.ts [hunk 3]
```diff
@@ -18,8 +18,8 @@ import { CmsComponentData } from '../../cms/components/cms-component-data';
 import { RoutingService } from '@spartacus/core';
 
 const UseCmsModuleConfig: CmsModuleConfig = {
-  cmsComponentMapping: {
-    SearchBoxComponent: 'SearchBoxComponent'
+  cmsComponents: {
+    SearchBoxComponent: { selector: 'SearchBoxComponent' }
```
Comment (tobi-or-not-tobi) [inline projects/storefrontlib/src/lib/cms-lib/search-box/search-box.component.spec.ts]: I'd expect this to be `cx-searchbox`.

File: tsconfig.json [hunk 4]
```diff
@@ -12,8 +12,8 @@
     "typeRoots": ["node_modules/@types"],
     "lib": ["es2015", "dom"],
     "strictPropertyInitialization": false,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
+//    "noUnusedLocals": true,
```
Comment (tobi-or-not-tobi) [inline tsconfig.json]: Did you intent to turn them off, or was it temporarily?

File: projects/core/src/site-context/facade/language.service.ts (hunk 2 above)
Comment (dunqan) [reply projects/core/src/site-context/facade/language.service.ts]: removed ;-)

**Pull request merged**